src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala

```diff
@@ -104,8 +104,18 @@ class JavaUrlConnectionRequestForwarder(
     leaderConnection.addRequestProperty(HEADER_FORWARDED_FOR, forwardedFor)
   }
 
+  private def setPatchMethod(connection: HttpURLConnection): Unit = {
+    val methodsField = classOf[HttpURLConnection].getDeclaredField("method")
+    methodsField.setAccessible(true)
```

Comment:
do we have to set it accessible every time? Or just once?

I think I'd like to see an object that encapsulates this unfortunate but necessary workaround

```
/**
 * Workaround to overcome Java restrictions; see https://bugs.openjdk.java.net/browse/JDK-7016595
 */
object HttpUrlConnectionWorkaround {
  val methodsField = classOf[HttpURLConnection].getDeclaredField("method")
  methodsField.setAccessible(true)

  def setMethod(connection: HttpURLConnection, method: String): Unit = {
    if (method == "PATCH")
      methodsField.set(connection, method)
    else
      connection.setRequestMethod(method)
  }
}
```